mautrix_telegram/__main__.py

```diff
@@ -189,6 +193,23 @@ async def _loop_active_puppet_metric(self) -> None:
                 return
             except Exception as e:
                 self.log.exception(f"Error while checking: {e}")
+
+    async def _loop_check_as_connection_pool(self) -> None:
+        while True:
+            try:
+                # a horrible reach into Appservice's internal API
+                connector = self.az._http_session.connector
+                limit = connector.limit
+                # a horrible, horrible reach into asyncio.TCPConnector's internal API
+                # inspired by its (also private) _available_connections()
+                active = len(connector._acquired)
+
+                METRIC_AS_CONNECTIONS.labels('active').set(active)
+                METRIC_AS_CONNECTIONS.labels('limit').set(limit)
+            except Exception as e:
+                self.log.exception(f"Error while checking AS connection pool stats: {e}")
+
+            await asyncio.sleep(1)
```

Comment:
Prometheus only polls every 15s afaik, so there is probably no point in being this hyper accurate.

Reply:
That being the case, perhaps we should apply a similar (more generically named) timeout for both this and RMAU?